### ISSUE TITLE
Keep getting this access violation tracebck: libgtk-3-0.dll!gtk_viewport_get_view_window

### ISSUE THREAD

**Issue by StevenWConley** [darktable-org/darktable#9802]
I am new to darktable - really want it to work - as I find the other alternative: (On1, Lightroom, etc) lacking in key points.
Biggest issue is a "proprietary" saving of adjustments.

But I am having frequent crashes now that I have started using it for "real" work developing my family photos.
(From Canon 60D, Sony RX100M6 and two different cell phones).

It seems from looking at crash files, more than 50% of crashes are the one illustrated here.

Doing VERY different things - mainly trying things trying to learn.  Just adjusting color or Filmic or what ever.

I am a programmer myself (retired from 49 years of it), and I know it is frustrating when no "steps to reproduce" are supplied.

The machine quoted below is very reliable at running and very very rarely does a program crash.

[darktable_bt_5P8870.txt](https://github.com/darktable-org/darktable/files/6979388/darktable_bt_5P8870.txt)
[darktable_bt_XNW770.txt](https://github.com/darktable-org/darktable/files/6979389/darktable_bt_XNW770.txt)

**Platform**

* darktable version : darktable-3.6.0.1-win64
* OS                :  Win7Pro SP1 (build 7601) with all to date patches.
* Memory            :  16GB
* Graphics card     :   2047MB NVIDIA GeForce GT 710 (ZOTAC International)
* Graphics driver   :  NVIDIA 2.1 Build_224 or   Date 8/12/2020, Driver Version 27.21.14.5206
* OpenCL installed  :  1.2 CUDA Driver 452.06
* OpenCL activated  :  Yes
* Desktop           :  Dell Optiplex 5040 with Intel Core i7-6700 @ 3.4GHz
* Display 2560 x 1440 @ 75MHz - 32bpp
**Additional context**

Although I used 3.4 just to see if it ran, I have been using 3.6 since it came out.

I work, in general with RAW files (CR2 or ARW) - the crashes have happened with both.

 Are the steps above reproducible with a fresh edit (i.e. after discarding history)? both ways

 Do you use lua scripts? no

**Comment by johnny-bit** [comment]
TBH this looks more like a GTK bug than darktable bug. The error in question and steps leading to it are some 5 levels of calls away from darktable core...

Would you be able to build yourself using windows build steps using `RelWithDebInfo` build type? That could give us some backtrace that we could work with.

**Comment by StevenWConley** [comment]
After poking around a while, I found build directions for Windows at
darktable <https://github.com/darktable-org/darktable>/packaging
<https://github.com/darktable-org/darktable/tree/master/packaging>/windows
<https://github.com/darktable-org/darktable/tree/master/packaging/windows>/
BUILD.md
on Github.

It looks like it will take a bit to pull it all together but I will give it
a try ... who knows - I "might" find the bug myself.
I have only done a "pseudo linux" compile on a Win machine one other time -
but I did get it working.

Alternatively - I do have 4 Ubuntu machines ... BUT none have the required
horsepower to run darktable - and worse if this is GTK it is perhaps
dependent on my exact video card (!!!).

Thanks for looking at my request there ... It will probably be a week
before I can pull this off.

Steve
